tests/testthat/test-workflow-bootstrap.R

```diff
@@ -0,0 +1,302 @@
+context("testing bootstrap functionality and running bbi")
+
+####################################################
+# testing bootstrap submission and summary via a composable workflow
+# This test file actually runs the bbi calls
+# and so it must have working version of both bbi and NONMEM.
+# Because of this, it is disabled unless on Metworx.
+#
+# Additionally, tests in this file rely on each other
+# and therefore must be run in order.
+####################################################
+
+# can't run on Drone because there's no NONMEM
+if (Sys.getenv("METWORX_VERSION") == "" || Sys.getenv("SKIP_BBI_TEST") == "true") {
+  skip("test-bootstrap only runs on Metworx because it needs NONMEM installed")
+}
+skip_long_tests("skipping long-running bbi workflow tests (bootstrap)")
+
+# define constants
+MODEL_DIR_BBI <- file.path(dirname(ABS_MODEL_DIR), "test-bootstrap")
+
+# cleanup function
+cleanup_bbi <- function(.recreate_dir = FALSE) {
+  if (fs::dir_exists(MODEL_DIR_BBI)) fs::dir_delete(MODEL_DIR_BBI)
+  if (isTRUE(.recreate_dir)) fs::dir_create(MODEL_DIR_BBI)
+}
+cleanup_bbi(.recreate_dir = TRUE)
+
+# set options and run tests
+withr::with_options(
+  list(
+    bbr.bbi_exe_path = read_bbi_path(),
+    bbr.verbose = TRUE
+  ), {
+
+    # cleanup when done
+    on.exit({
+      Sys.sleep(3) # wait for some NONMEM mess to delete itself
+      cleanup_bbi()
+    })
+
+    # clear old bbi.yaml
+    if (fs::file_exists(file.path(MODEL_DIR_BBI, "bbi.yaml"))){
+      fs::file_delete(file.path(MODEL_DIR_BBI, "bbi.yaml"))
+    }
+
+    # create new bbi.yaml
+    bbi_init(
+      MODEL_DIR_BBI,
+      .nonmem_dir = Sys.getenv("BBR_TESTS_NONMEM_DIR", "/opt/NONMEM"),
+      .nonmem_version = Sys.getenv("BBR_TESTS_NONMEM_VERSION", "nm74gf"),
+      .bbi_args = list(mpi_exec_path = get_mpiexec_path())
+    )
+
+    # copy model file into new model dir
+    fs::file_copy(CTL_TEST_FILE, MODEL_DIR_BBI)
+
+    # Create base model to bootstrap with (not yet submitted)
+    mod1 <- new_model(
+      file.path(MODEL_DIR_BBI, "1"),
+      .description = "original test-bootstrap model",
+      .tags = ORIG_TAGS,
+      .bbi_args = list(threads = 4, parallel = TRUE)
```

Comment:
These simple models sometimes fail when parallelized. Unless there's a reason to keep them, I would remove these `.bbi_args`, just to avoid flakiness.